runners/core-java/src/test/java/org/apache/beam/runners/core/SimpleDoFnRunnerTest.java

```diff
@@ -234,6 +239,97 @@ public void testOnTimerCalled() {
                 TimeDomain.EVENT_TIME)));
   }
 
+  @Test
+  public void testBackwardsInTimeNoSkew() {
```

Comment:
Add short comments to all of these tests what they're testing.
E.g. this one is testing that with an allowed skew of 0, you can't shift backwards at all. The next one tests actual skew > allowed skew (non-zero), etc.